### PR TITLE
fix, refactor, chore(egen): removes keras3 dependency error while saving the model, refactors code as per template

### PR DESCRIPTION
**Pull request by sumanvita-springml**
This PR includes the following changes:

1. In the training component, changes base image from ""tensorflow/tensorflow:latest" to "tensorflow/tensorflow" to download last stable version of TF.
2. Changes model.save() to tf.saved_model.save() to resolve keras3 dependency issue
3. Changes REGION to LOCATION
4. Includes colab restart kernal and authentication as per template guidelines
5. Adds code to delete custom job in cleanup section
6. Changes the order of Github and workbench logos
7. Made structural changes according to the new notebook template

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

**Review by kittyabs — COMMENTED** [review]
no edits needed

File: notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb [hunk 1]
```diff
@@ -921,14 +831,28 @@
...
-        "\n",
         "# Delete TensorBoard instance.\n",
         "! gcloud ai tensorboards delete {TENSORBOARD_RESOURCE_NAME}\n",
         "\n",
-        "# Delete custom job.\n",
-        "job.delete()"
+        "# Delete Vertex AI pipeline.\n",
+        "job.delete()\n",
+        "\n",
+        "# Delete the CustomJob using its display name.\n",
```
Comment (kittyabs) [inline notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb:840]: backticks for `CustomJob` ??

File: notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb [hunk 2]
```diff
@@ -921,14 +831,28 @@
...
+        "jobs = aiplatform.CustomJob.list(filter='display_name=\"Trainer\"')\n",
+        "if jobs:\n",
+        "    job = jobs[0]\n",
+        "    job.delete()\n",
+        "    print(\"Deleted CustomJob:\", job.display_name)\n",
+        "else:\n",
+        "    print(\"No CustomJob found with the display name 'Trainer'\")\n",
```
Comment (kittyabs) [inline notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb:847]: backticks for `CustomJob `??

**Review by kittyabs — COMMENTED** [review]
suggested edits

File: notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb (hunk 1 above)
Comment (sumanvita-springml) [reply notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb:840]: It is inside print statement

File: notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb (hunk 2 above)
Comment (sumanvita-springml) [reply notebooks/official/tensorboard/tensorboard_vertex_ai_pipelines_integration.ipynb:847]: It's inside print statement. Removed the print statements.

**Review by katiemn — APPROVED** [review]

**Pull request merged**